### PR TITLE
feat: update cards body to fit with the new layout

### PR DESCRIPTION
**Pull request by DragosIorgulescu**
## Changes

- covered welcome, share, article & ad card types

Notes:
- ~~squad feed layout seems broken~~
- ~~mobileXL not being picked up properly~~

## Preview

**Article & Ad cards** 
Screenshot 2024-01-15 at 12 49 24
Screenshot 2024-01-15 at 12 49 12

**Freeform & Share cards**
Screenshot 2024-01-15 at 12 49 46

**Mobile**

https://github.com/dailydotdev/apps/assets/1225100/8526cfab-1377-43d9-ac5d-3e327e6711dd

## Events

N / A

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

MI-16 #done

### CODE REVIEW

File: packages/shared/src/components/cards/v1/ArticlePostCard.tsx [hunk 1]
```diff
@@ -94,26 +104,39 @@ export const ArticlePostCard = forwardRef(function PostCard(
...
-            </CardTitle>
-          </CardTextContainer>
-          {post.summary && (
-            <CardTextContainer className="mt-4 text-theme-label-secondary">
-              {post.summary}
-            </CardTextContainer>
-          )}
+
+            <CardContent>
+              <div className="mr-4 flex-shrink flex-grow">
```
Comment (rebelchris) [inline packages/shared/src/components/cards/v1/ArticlePostCard.tsx:109]: Would `flex-1` no do? (instead of both flex-shrink flex-grow)

File: packages/shared/src/components/cards/v1/FeedbackCard.tsx [hunk 2]
```diff
@@ -10,7 +10,7 @@ import UpvoteIcon from '../../icons/Upvote';
 import { Post, UserPostVote } from '../../../graphql/posts';
 import { usePostFeedback } from '../../../hooks';
 import CloseButton from '../../CloseButton';
-import { PostCardFooter } from './PostCardFooter';
+// import { PostCardFooter } from './PostCardFooter';
```
Comment (rebelchris) [inline packages/shared/src/components/cards/v1/FeedbackCard.tsx:13]: Removed, or another ticket?
If so please mention ticket TODO

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 3]
```diff
@@ -66,35 +80,58 @@ export const WelcomePostCard = forwardRef(function SharePostCard(
...
+        >
+          <SquadHeaderPicture
+            author={post.author}
+            source={post.source}
+            reverse={!enableSourceHeader}
+          />
+        </PostCardHeader>
+
+        <CardContent>
+          <div className="mr-4 flex-shrink flex-grow">
```
Comment (rebelchris) [inline packages/shared/src/components/cards/v1/WelcomePostCard.tsx:102]: Same flex-1 thing

File: packages/shared/tailwind.config.js [hunk 4]
```diff
@@ -154,6 +154,7 @@ module.exports = {
     },
     screens: {
       mobileL: '420px',
+      mobileXL: '500px',
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/tailwind.config.js:157]: Why do we need this new row?
Is it that this design introduces a new breakpoint?

(If so could you point me to the design issue, it seems inconsistent)

**Review by rebelchris — COMMENTED** [review]
Minor comments, nothing blocking from my side.

File: packages/shared/tailwind.config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DragosIorgulescu) [reply packages/shared/tailwind.config.js:157]: @rebelchris the spec specifies this new breakpoint, indeed, you can find the info here: https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/681410561/WT-1957+Feed+Layout+Improvements#Main

File: packages/shared/src/components/cards/v1/FeedbackCard.tsx (hunk 2 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/FeedbackCard.tsx:13]: good catch, forgot to remove these

File: packages/shared/tailwind.config.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/tailwind.config.js:157]: I see, so to make it work you can have it there.
It's probably tailwind caching, so basically when you add some tailwind new values you need to clear next cache.

Inside webapp remove the `.next` folder and build again, then it should work :)

File: packages/shared/src/components/cards/SharedPostCardFooter.tsx [hunk 5]
```diff
@@ -20,26 +20,29 @@ export const SharedPostCardFooter = ({
     <div
```
Comment (nensidosari) [inline packages/shared/src/components/cards/SharedPostCardFooter.tsx:20]: Should all the changes in this file be behind the experiment? 
If yes - I think we should copy paste this file in the v1 folder within cards which contains all the files that get changed conditionally by the feed layout experiment.

File: packages/shared/src/components/cards/WelcomePostCardFooter.tsx [hunk 6]
```diff
@@ -39,7 +39,7 @@ export const WelcomePostCardFooter = ({
```
Comment (nensidosari) [inline packages/shared/src/components/cards/WelcomePostCardFooter.tsx:39]: Same here, checking if these changes should be behind the experiment?

File: packages/shared/src/components/cards/v1/ArticlePostCard.tsx [hunk 7]
```diff
@@ -94,26 +104,39 @@ export const ArticlePostCard = forwardRef(function PostCard(
...
+                >
+                  {post.title}
+                </CardTitle>
+
+                {post.summary && (
+                  <CardTextContainer className="mt-4 text-theme-label-secondary">
+                    {post.summary}
+                  </CardTextContainer>
+                )}
+              </div>
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/ArticlePostCard.tsx:122]: Are the title and summary limits set based on the notes on DR? https://dailydotdev.atlassian.net/wiki/spaces/HAN/pages/681410561/WT-1957+Feed+Layout+Improvements#Main

Ex of one of the notes for limits:
All posts (Title + Tldr) will be limited to 300 chars total

File: packages/shared/src/components/cards/v1/FeedbackCard.tsx [hunk 8]
```diff
@@ -73,15 +72,15 @@ export const FeedbackCard = ({
...
-        />
+        {/* <PostCardFooter */}
+        {/*  className={{ */}
+        {/*    image: 'mb-0', */}
+        {/*  }} */}
+        {/*  post={post} */}
+        {/*  showImage={showImage} */}
+        {/*  insaneMode={false} */}
+        {/*  openNewTab */}
+        {/* /> */}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/FeedbackCard.tsx:83]: Should this be removed or properly added?

**Review by nensidosari — CHANGES_REQUESTED** [review]
Some concerning comments on my side on terms of experiment and the text limits, otherwise looks good.

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 9]
```diff
@@ -47,6 +49,18 @@ export const WelcomePostCard = forwardRef(function SharePostCard(
       openStep,
     );
 
+  const content = useMemo(
+    () =>
+      post.contentHtml ? sanitize(post.contentHtml, { ALLOWED_TAGS: [] }) : '',
+    [post.contentHtml],
+  );
```
[the remark this answers is not in the document]
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:56]: I just copied the code we already had handling this in the original card components, if we want to clean check and clean this up, I propose we do it as a separate task. Thoughts?

File: packages/shared/src/components/cards/SharedPostCardFooter.tsx (hunk 5 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/SharedPostCardFooter.tsx:20]: great catch, my bad, yeah, this should be copied under V1 👍

File: packages/shared/src/components/cards/SharedPostCardFooter.tsx [hunk 10]
```diff
@@ -20,26 +20,29 @@ export const SharedPostCardFooter = ({
...
       className={classNames(
         'mb-2 flex h-auto min-h-0 w-full flex-auto gap-3 rounded-12 border border-theme-divider-tertiary p-3',
-        isShort ? 'flex-row items-center' : 'flex-col',
+        isShort ? 'flex-row items-center' : 'flex-col  mobileL:flex-row',
```
[the remark this answers is not in the document]
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/SharedPostCardFooter.tsx:23]: yes, it was not working for me so I used `mobileL` to test, will switch to `mobileXL`

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (rebelchris) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:56]: Afaik we decided this for welcome posts as they render inline already.
We still do sanitisation backend, this is more for the edit capabilities.
Correct me if I'm wrong @sshanzel

File: packages/shared/src/components/cards/v1/ArticlePostCard.tsx (hunk 7 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/ArticlePostCard.tsx]: Done, thank you for pointing it out

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:56]: > I just copied the code we already had handling this in the original card components

If we're reusing maybe useful to add in a utils file or sth to use for both, but maybe for this phase is fine either way.

**Review by rebelchris — APPROVED** [review]
Nothing blocking from my side.
Didn't explicitly looked at design results though (keeping it for final review)

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+        isShort ? 'flex-row items-center' : 'flex-col  mobileL:flex-row',
+      )}
+    >
+      <div
+        className={classNames(
+          'flex flex-col mobileL:flex-1',
+          isShort ? 'line-clamp-4 w-8/12 pr-3' : 'line-clamp-2',
+        )}
+      >
+        {sharedPost.title}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx:32]: Just making sure, in the notes this title is said to be max 1 line, so I'm guessing this needs to always be line-clamp-1

Comment (nensidosari) [reply packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx:32]: How are we using is Short in this case, is it needed with the new designs?

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+        <span className="text-theme-label-quaternary">
+          {sharedPost.summary}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx:34]: Just making sure, in the notes this title is said to be max 2 line, so I'm guessing this needs to always be line-clamp-2?

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx [hunk 13]
```diff
@@ -0,0 +1,55 @@
...
+        </span>
+      </div>
+
+      <div className={classNames('flex h-auto overflow-auto')}>
+        <ImageComponent
+          size={isShort ? IconSize.XLarge : IconSize.XXXLarge}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx:40]: Same here, How are we using is Short in this case, is it needed with the new designs?

File: packages/shared/src/components/cards/v1/SharePostCard.tsx [hunk 14]
```diff
@@ -1,7 +1,7 @@
 import React, { forwardRef, ReactElement, Ref, useRef, useState } from 'react';
 import ActionButtons from './ActionButtons';
 import { SharedPostText } from '../SharedPostText';
-import { SharedPostCardFooter } from '../SharedPostCardFooter';
+import { SharedPostCardFooter } from './SharedPostCardFooter';
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/SharePostCard.tsx:4]: Not on this line, but further down, can't comment there, we need to make sure we follow the rule for summary to not be more then 300 chars here as well.

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:56]: @DragosIorgulescu, maybe forgotten, though non blocking, would we be moving this?

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 15]
```diff
@@ -66,35 +80,58 @@ export const WelcomePostCard = forwardRef(function SharePostCard(
...
+            <CardTitle
+              className={classNames(
+                generateTitleClamp({
+                  hasImage: !!image,
+                  hasHtmlContent: !!post.contentHtml,
+                }),
+                'multi-truncate',
+              )}
+            >
+              {post.title}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/WelcomePostCard.tsx:112]: I believe we should be using this here as well  `const { title, summary } = useTruncatedSummary(post);`

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 16]
```diff
@@ -66,35 +80,58 @@ export const WelcomePostCard = forwardRef(function SharePostCard(
...
+            </CardTitle>
+
+            {content && (
+              <p className="my-4 line-clamp-6 break-words typo-callout">
+                {decodedText}
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/WelcomePostCard.tsx:117]: I believe we should be using this here as well `const { title, summary } = useTruncatedSummary(post);`

**Review by nensidosari — CHANGES_REQUESTED** [review]
Looks great overall, few more things on my side that I thing need to be changed. Please take a look, should be small stuff 🙏

**Comment by nensidosari** [comment]
@Dragos few more things noticed when checking on designs:

- This is not on your code or ticket, but since we're here, if u search for `laptop:px-12` in code u should find that in a `<Feed...` component being rendered, that should be `laptop:px-6` actually. It has been missed, if u could change it would be great.

- Another thing on the video youtube overlay, this shows in the feed:
Screenshot 2024-01-16 at 11 43 44 AM

- Last thing, are u still working on adding the 550px breakpoint with smaller image? doesn't seem to be on the code yet, it was missed from the specs though we are aware of it.

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx (hunk 11 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx]: I removed the usage of `isShort` from these, as it wasn't being used. Great catch 🙏

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 16 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:119]: the decoded text will be a bit trickier because it is potentially HTML, and we just truncated it based on length we run the risk of getting broken HTML. I suggest we merge this as is (so we can publish and start doing testing) while we tackle this particular task separately?

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx (hunk 13 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx]: removed `isShort` from here as well, not needed for these designs

**Comment by DragosIorgulescu** [comment]
> This is not on your code or ticket, but since we're here, if u search for laptop:px-12 in code u should find that in a <Feed... component being rendered, that should be laptop:px-6 actually. It has been missed, if u could change it would be great.

done 👍 

> Last thing, are u still working on adding the 550px breakpoint with smaller image? doesn't seem to be on the code yet, it was missed from the specs though we are aware of it.

done as well 🙏

File: packages/shared/src/components/cards/v1/AdCard.tsx [hunk 17]
```diff
@@ -45,7 +45,7 @@ export const AdCard = forwardRef(function AdCard(
               alt="Ad image"
               src={ad.image}
               className={classNames(
-                'z-1 w-full mobileXL:w-56',
+                'z-1 w-full mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56',
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/AdCard.tsx:48]: I see this `mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56` is now in all CardImage component, correct me if I'm wrong. If yes, can we possibly make the change there to be in just one place?

File: packages/shared/src/components/cards/v1/AdCard.tsx [hunk 18]
```diff
@@ -54,7 +54,7 @@ export const AdCard = forwardRef(function AdCard(
               <CardImage
                 alt="Ad image background"
                 src={ad.image}
-                className="-z-1 w-full blur-20 mobileXL:w-56"
+                className="-z-1 w-full blur-20 mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56"
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/AdCard.tsx:57]: Same as above

File: packages/shared/src/components/cards/v1/ArticlePostCard.tsx [hunk 19]
```diff
@@ -131,7 +131,7 @@ export const ArticlePostCard = forwardRef(function PostCard(
                 src={post.image}
                 fallbackSrc={cloudinary.post.imageCoverPlaceholder}
                 className={classNames(
-                  'object-cover mobileXL:w-56',
+                  'object-cover mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56',
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/ArticlePostCard.tsx:134]: Same as above

File: packages/shared/src/components/cards/v1/CollectionCard.tsx [hunk 20]
```diff
@@ -90,7 +90,7 @@ export const CollectionCard = forwardRef(function CollectionCard(
               alt="Post Cover image"
               src={image}
               fallbackSrc={cloudinary.post.imageCoverPlaceholder}
-              className="my-2 object-cover mobileXL:w-56"
+              className="my-2 object-cover mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56"
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/CollectionCard.tsx:93]: Same as above

**Comment by DragosIorgulescu** [comment]
@nensidosari I made all the neccessary edits, with one exception:

> Another thing on the video youtube overlay, this shows in the feed:

I cannot seem to be able to enable videos locally, so I can't get my feed to display these card types 🤔

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx [hunk 21]
```diff
@@ -6,44 +6,35 @@ import { IconSize } from '../../Icon';
...
     >
-      <div
-        className={classNames(
-          'flex flex-col mobileL:flex-1',
-          isShort ? 'line-clamp-4 w-8/12 pr-3' : 'line-clamp-2',
-        )}
-      >
-        {sharedPost.title}
-        <span className="text-theme-label-quaternary">
+      <div className={classNames('line-clamp-1 flex flex-col mobileL:flex-1')}>
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx:23]: Why would the wrapping div have a `line-clamp-1` if the children have different line clamps?

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 22]
```diff
@@ -124,7 +126,7 @@ export const WelcomePostCard = forwardRef(function SharePostCard(
               alt="Post Cover image"
               src={image}
               fallbackSrc={cloudinary.post.imageCoverPlaceholder}
-              className="my-2 object-cover mobileXL:w-56"
+              className="my-2 object-cover mobileXL:h-28 mobileXL:w-40 mobileXXL:h-auto mobileXXL:w-56"
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/WelcomePostCard.tsx:129]: Same as comments above

**Review by nensidosari — COMMENTED** [review]
Few more stuff 🙈 🙏

**Comment by nensidosari** [comment]
> I cannot seem to be able to enable videos locally, so I can't get my feed to display these card types 🤔
I know that's a pain already 🙈 If you can please modify any of the cards on your feed to be of type video on DB. Can help u do that if u need my help 🙏

File: packages/shared/src/components/cards/v1/AdCard.tsx (hunk 17 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/AdCard.tsx]: done 🙏

File: packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx (hunk 21 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/SharedPostCardFooter.tsx]: this class applied to the div is actually completely ignored -- removed 🙏

File: packages/shared/src/components/image/VideoImage.tsx [hunk 23]
```diff
@@ -18,7 +18,7 @@ const VideoImage = ({
     <div
       className={classNames(
         wrapperClassName,
-        'relative flex h-auto w-full items-center justify-center rounded-12',
+        'relative flex h-auto items-center justify-center rounded-12',
```
Comment (DragosIorgulescu) [inline packages/shared/src/components/image/VideoImage.tsx:21]: @nensidosari this fixes the video thumbnail sizing issue as well, it also does not affect the previous layout cards based on my testing, so we should not need to copy this component to `v1` just for this change.

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 16 above)
Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:118]: So the decoded text is not just text? Just concerned if we don't add this it might mess with designs and requirements...
Still html or not, if we're getting the innerText that should have characters, and we would be able to check the length no?

File: packages/shared/src/components/image/VideoImage.tsx (hunk 23 above)
Comment (nensidosari) [reply packages/shared/src/components/image/VideoImage.tsx:21]: Ah, got the issue, it messes with the video types on shared posts

Comment (nensidosari) [reply packages/shared/src/components/image/VideoImage.tsx:21]: We'll need to fix @DragosIorgulescu

Comment (DragosIorgulescu) [reply packages/shared/src/components/image/VideoImage.tsx]: updated so we only apply width adjustments on the new layout

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 16 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:118]: this is built based on `post.contentHtml` which comes in the following format: 
```
<p>Welcome to test, a dedicated space to collaborate, share knowledge, and discuss topics that matter to us!</p>
<p>Here are some of the things you can do in Squads:</p>
<ul>
<li>Say hi: Start by saying hi in the comments below so that we’ll know you’re here</li>
<li>Create a new post: Share interesting links and your thoughts by creating a new post in the squad</li>
<li>Interact with others: Comment and upvote on other members’ posts and give feedback</li>
<li>Personalize it: Customize your profile by adding a profile picture, bio, and links to your projects or social media</li>
<li>Invite other developers you know and appreciate that you think can benefit from this squad</li>
</ul>
<p>Now that you know what you can do in this squad, we’ve put together a code of conduct that we expect all of our squad members to follow:</p>
<ol>
<li>Keep it relevant: Keep your posts and comments relevant to the topic of the Squad. Please refrain from spamming or promoting unrelated content.</li>
<li>Be respectful: Treat others the way you want to be treated. We do not tolerate hate speech, discrimination, or harassment of any kind.</li>
<li>Be constructive: Offer helpful feedback and constructive criticism rather than tearing others down.</li>
<li>Protect your privacy: Do not share personal information or sensitive data in Squads.</li>
</ol>
<p>We hope you will find test useful!</p>
```

This is why truncating this is going to be a bit more complex. We have the `line-clamp-6` as a stopgap to ensure this does not overflow out of control, but I propose we tackle this separately if we want to really enforce a 300 character limit on this.

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:57]: Agreed to leave this for now, and possibly refactor once the experiment is concluded.

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 16 above)
Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:118]: Cool then if we have the `line-clamp-6` I think we should be fine for this not to overflow. Maybe would be good to just note this in the channel and tag design that we're moving on like this for the welcome card because of this possible issue. @DragosIorgulescu would u mind sending a message?

**Review by nensidosari — APPROVED** [review]
Looks great now! No further comments. 🥳

File: packages/shared/tailwind.config.js [hunk 24]
```diff
@@ -154,6 +154,8 @@ module.exports = {
...
+      mobileXXL: '550px',
```
Comment (rebelchris) [inline packages/shared/tailwind.config.js:158]: Ugh, not your fault but this is pain to my eyes 🤣 
@nensidosari Is the idea to eventually remove/change the breakpoints once we find experiment winner?

**Review by rebelchris — APPROVED** [review]
Nothing blocking from my side.
Nice work!

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx [hunk 25]
```diff
@@ -1,18 +1,19 @@
-import React, { forwardRef, ReactElement, Ref, useRef } from 'react';
+import React, { forwardRef, ReactElement, Ref, useMemo, useRef } from 'react';
```
Comment (nensidosari) [inline packages/shared/src/components/cards/v1/WelcomePostCard.tsx:1]: Just realised we've been touching welcome post card unneccissarily, this is not needed for the feed layout changes. Let's revert the changes here and in useTruncatedSummary made specifically for this use case 🙏

**Review by nensidosari — CHANGES_REQUESTED** [review]
Last thing I just realised, we should be good with all other things. Once we fix this we're good to go

File: packages/shared/tailwind.config.js (hunk 24 above)
Comment (nensidosari) [reply packages/shared/tailwind.config.js:158]: The idea was to have these for the experiment, depending on how that goes yes they will be removed or kept. Maybe we should leave a todo note here to remember to remove these if the experiment is not a winner. 

@DragosIorgulescu can u add the TODO comment?

File: packages/shared/src/components/cards/v1/WelcomePostCard.tsx (hunk 25 above)
Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:1]: Damn, great point 😅 I completely forgot that these changes will not be applied to squad feeds... 🤦🏻‍♂️ 

Will revert, but I think we can keep the welcome card around even if not used

Comment (DragosIorgulescu) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:1]: we actually use the welcome card for freeform post types too: https://github.com/dailydotdev/apps/blob/1fbc96daf98d806b557ab63f55f37af3a9361674/packages/shared/src/components/FeedItemComponent.tsx#L105 -- so maybe we should keep it around, since these post types may show up in non-squad feeds

Comment (nensidosari) [reply packages/shared/src/components/cards/v1/WelcomePostCard.tsx:1]: Ah ok if so yes, all should be good

**Review by nensidosari — COMMENTED** [review]
Merged now, so no approval. Just notting this was approved and good to merge!
